packages/story-editor/src/elements/video/trim.js

```diff
@@ -59,38 +69,73 @@ function VideoTrim({ box: { width, height }, element }) {
 
   videoProps.crossOrigin = 'anonymous';
 
+  const boxAtOrigin = useMemo(
+    () => ({
+      ...box,
+      x: 0,
+      y: 0,
+    }),
+    [box]
+  );
+
+  const { setVideoNode } = useVideoTrim(({ actions: { setVideoNode } }) => ({
+    setVideoNode,
+  }));
+  const setRef = useCallback(
+    (node) => {
+      videoRef.current = node;
+      setVideoNode(node);
+    },
+    [setVideoNode]
+  );
+
   return (
-    <MediaDisplay
-      element={element}
-      mediaRef={ref}
-      showPlaceholder
-      previewMode={false}
-    >
-      <Video
-        poster={poster || resource.poster}
-        style={style}
-        {...videoProps}
-        preload="metadata"
-        loop
-        muted
-        autoPlay
-        tabIndex={0}
-        ref={ref}
-      >
-        {resource.src && <source src={resource.src} type={resource.mimeType} />}
-        {tracks &&
-          tracks.map(({ srclang, label, kind, track: src, id: key }, i) => (
-            <track
-              srcLang={srclang}
-              label={label}
-              kind={kind}
-              src={src}
-              key={key}
-              default={i === 0}
-            />
-          ))}
-      </Video>
-    </MediaDisplay>
+    <>
+      <Wrapper ref={wrapperRef}>
+        <MediaDisplay
+          element={element}
+          mediaRef={videoRef}
+          showPlaceholder
+          previewMode={false}
+        >
+          <Video
+            poster={poster || resource.poster}
+            style={style}
+            {...videoProps}
+            preload="metadata"
+            muted
+            autoPlay
+            tabIndex={0}
+            ref={setRef}
+          >
+            {resource.src && (
+              <source src={resource.src} type={resource.mimeType} />
+            )}
+            {tracks &&
+              tracks.map(({ srclang, label, kind, track: src, id: key }, i) => (
+                <track
+                  srcLang={srclang}
+                  label={label}
+                  kind={kind}
+                  src={src}
+                  key={key}
+                  default={i === 0}
+                />
+              ))}
+          </Video>
+        </MediaDisplay>
+      </Wrapper>
+      <VideoControls
+        isEditing={false}
+        isTransforming={false}
+        isSingleElement
+        isSelected
+        box={boxAtOrigin}
```

Comment:
This is super ugly and should be cleaned up. Why set all these variables to some magic values just to get the component to appear? Much better to create an extra component and use that in the two places where relevant.

Reply:
Done in 04d0731ea6.